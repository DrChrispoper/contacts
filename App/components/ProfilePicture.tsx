import React from 'react';
import { StyleSheet, Image, TouchableOpacity } from 'react-native';
import { Profile } from '../utils/types';

const styles = StyleSheet.create({
  item: {
    padding: 0,
    marginVertical: 8,
    marginHorizontal: 8,
    width: 70,
    height: 70,
  },
  active: {
    overflow: 'hidden',
    borderWidth: 5,
    borderColor: '#b1d8ff',
    borderRadius: 70 / 2,
  },
  tempFix: {
    transform: [{ rotate: '90deg' }],
  },
});

interface Props {
  item: Profile;
  onPress?: () => void;
  isSelected?: boolean;
  inDetailsPage?: boolean;
}

class ProfilePicture extends React.PureComponent<Props> {
  render(): JSX.Element {
    const { item, onPress, isSelected = false, inDetailsPage = false } = this.props;
    return (
      <TouchableOpacity
        onPress={onPress}
        style={[styles.item, isSelected && styles.active, !inDetailsPage && styles.tempFix]}
      >
        <Image source={item.picture} />
      </TouchableOpacity>
    );
  }
}

export default ProfilePicture;
