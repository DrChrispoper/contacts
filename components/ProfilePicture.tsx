import React from 'react';
import { StyleSheet, Image, TouchableOpacity, ViewStyle } from 'react-native';
import { Profile } from '../utils/types';
import { Allan } from '../utils/imageLoader';

const styles = StyleSheet.create({
  item: {
    padding: 0,
    marginVertical: 8,
    marginHorizontal: 16,
    width: 70,
    height: 70,
  },
  active: {
    overflow: 'hidden',
    borderWidth: 5,
    borderColor: '#b1d8ff',
    borderRadius: 150 / 2,
  },
});

interface Props {
  item: Profile;
  onPress: () => void;
  isSelected: boolean;
}

const ProfilePicture: React.FC<Props> = ({ item, onPress, isSelected }: Props) => (
  <TouchableOpacity onPress={onPress} style={[styles.item, isSelected && styles.active]}>
    <Image source={item.picture} />
  </TouchableOpacity>
);

export default ProfilePicture;
