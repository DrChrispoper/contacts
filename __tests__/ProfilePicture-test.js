/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TouchableOpacity } from 'react-native';
import ProfilePicture from '../components/ProfilePicture';

Enzyme.configure({ adapter: new Adapter() });

describe('Profile Picture Component', () => {
  const profile = {
    id: 'DJKDSQLDQKS',
    fullName: 'Christopher Hockley',
    firstName: 'Christopher',
    lastName: 'Hockley',
    role: 'Developer',
    picture: jest.mock('../assets/images/avatars/Allan-Munger.png'),
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  };

  test('renders correctly', () => {
    const onPressEvent = jest.fn();

    const tree = renderer
      .create(<ProfilePicture item={profile} onPress={onPressEvent} isSelected={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles onPress', () => {
    // Create a mock function to pass as a handler
    const onPress = jest.fn();

    // Render our component
    const wrapper = shallow(<ProfilePicture item={profile} onPress={onPress} isSelected={false} />);
    // Find a TouchableOpacity and press it
    wrapper.find(TouchableOpacity).first().props().onPress();

    // Check that our handler have been called 1 time
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
