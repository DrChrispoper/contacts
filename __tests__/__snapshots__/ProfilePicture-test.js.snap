// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Picture Component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 70,
      "marginHorizontal": 8,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 0,
      "width": 70,
    }
  }
>
  <Image
    source={
      Object {
        "addMatchers": [Function],
        "advanceTimersByTime": [Function],
        "advanceTimersToNextTimer": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "getTimerCount": [Function],
        "isMockFunction": [Function],
        "isolateModules": [Function],
        "mock": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModuleRegistry": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "runTimersToTime": [Function],
        "setMock": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />
</View>
`;
